### PR TITLE
feature/FOUR-11507: [SPRING] Admin Settings Tab Reorganization

### PR DESCRIPTION
**Pull request by pmPaulis**
## Issue & Reproduction Steps
Admin Settings Tab Reorganization
**PRD** https://processmaker.atlassian.net/wiki/spaces/PM4/pages/3215917057/Admin+Settings+Tab+Reorganization
**FIGMA** https://www.figma.com/file/Tr2bVMtC4jr6KGWBvM6Vtp/Spring-24?type=design&node-id=863-45331&mode=design&t=nhGXOjrnVyIrHcnh-0

## Solution
Improve the Menus Reorganization for settings

## How to Test
- Go to Admin -> Settigns
- **Email**
- Email -> Action By Email
- Email -> Email Default Settings
- **Log-In Auth**
- Log-In Auth -> SSO
- Log-In Auth -> SSO - Atlassian
- Log-In Auth -> Auth0
- Log-In Auth -> Facebook
- Log-In Auth -> Github
- Log-In Auth -> Google
- Log-In Auth -> Keycloak
- Log-In Auth -> Ldap
- Log-In Auth -> Microdoft
- Log-In Auth ->  Saml
- Log-In Auth → Scim
- **User Settings**
- User Settings →Users
- **Integrations**
-  Integrations -> Docusign
- Integrations -> Idp
- Integrations -> External Integrations

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-11507

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next
ci:deploy
ci:connector-docusign:feature/FOUR-11507
ci:connector-idp:feature/FOUR-11507
ci:package-cdata:feature/FOUR-11507
ci:connector-send-email:feature/FOUR-11507
ci:package-actions-by-email:feature/FOUR-11507
ci:package-auth:feature/FOUR-11507
ci:package-advanced-user-manager:feature/FOUR-11507

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Review by fagubla — APPROVED** [review]

File: ProcessMaker/Models/Setting.php [hunk 1]
```diff
@@ -370,4 +372,104 @@ public static function getFavicon()
...
+            ->where('group_id', $menuId)
+            ->notHidden()
+            ->pluck('group');
+        $response = $query->toArray();
+        $result = [];
+        foreach ($response as &$value) {
+            // Technical debts: we need to add int key to identify a group, currently this is a label
+            $result[] = [
+                'id' => $value,
+                'name' => $value,
```
Comment (caleeli) [inline ProcessMaker/Models/Setting.php:397]: Consider to add a `group_label` field for the UI text.

```suggestion
                'name' => $value,
                'group_label' => $value,
```

File: resources/js/admin/settings/components/SettingsMenuCollapse.vue [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+              </span>
+              <i :class="{ 'fas fa-caret-down': collapsedMenus[menu.id], 'fas fa-caret-up': !collapsedMenus[menu.id] }" />
+            </div>
+          </h5>
+        </div>
+
+        <div
+          :id="`collapseOne${menu.id}`"
+          class="collapse"
+          :class="menu.menu_group_order === 1 ? 'show' : ''"
```
Comment (caleeli) [inline resources/js/admin/settings/components/SettingsMenuCollapse.vue:35]:
```suggestion
          :class="{ show: menu.menu_group_order === 1 }"
```

File: tests/Feature/Api/SettingsTest.php [hunk 3]
```diff
@@ -33,6 +35,89 @@ class SettingsTest extends TestCase
         'created_at',
     ];
 
+    /**
+     * Test extended properties variable valid name validation
```
Comment (caleeli) [inline tests/Feature/Api/SettingsTest.php:39]: Please fix the test description

File: tests/Feature/Api/SettingsTest.php [hunk 4]
```diff
@@ -33,6 +35,89 @@ class SettingsTest extends TestCase
...
+            'format' => 'test',
+            'group' => 'UserTest',
+            'group_id' => $menus->id,
+        ]);
+        $route = route('api.settings.menu_groups');
+        $response = $this->apiCall('GET', $route);
+        // Verify the status
+        $response->assertStatus(200);
+        $this->assertNotEmpty($response['data']);
+        $this->assertNotEmpty($response['data'][0]['groups']);
```
Comment (caleeli) [inline tests/Feature/Api/SettingsTest.php:71]: Consider to improve this assertion to verify the expected result

File: tests/Feature/Api/SettingsTest.php [hunk 5]
```diff
@@ -33,6 +35,89 @@ class SettingsTest extends TestCase
...
+            'name' => 'test',
+            'format' => 'test',
+            'group' => $group,
+            'group_id' => null,
+        ]);
+
+        // Update
+        Setting::updateSettingsGroup($group, $menus->id);
+        $matches = Setting::where('group', $group)->where('group_id', $menus->id)->get()->toArray();
+        $this->assertNotEmpty($matches);
```
Comment (caleeli) [inline tests/Feature/Api/SettingsTest.php:92]: Consider to improve the expected result

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-511b59539d.engk8s.processmaker.net

**Comment by pmPaulis** [comment]
@ryancooley @MauricioVeliz 
This PR require other changes in the following packages:
https://github.com/ProcessMaker/connector-idp/pull/39 
https://github.com/ProcessMaker/connector-send-email/pull/298 
https://github.com/ProcessMaker/package-actions-by-email/pull/127 
https://github.com/ProcessMaker/package-cdata/pull/32 
https://github.com/ProcessMaker/package-auth/pull/101 
https://github.com/ProcessMaker/package-advanced-user-manager/pull/41

All the PR was marked **FOR REVIEW**
Regards,

**Pull request merged**